service/controller/v22/resource/s3bucket/create.go

```diff
@@ -49,66 +46,97 @@ func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange inte
+					},
+				}
+
+				_, err = sc.AWSClient.S3.PutBucketTagging(i)
+				if err != nil {
+					return microerror.Mask(err)
+				}
+
+				return nil
+			})
 		}
 
 		if bucketInput.IsLoggingBucket {
-			_, err = sc.AWSClient.S3.PutBucketAcl(&s3.PutBucketAclInput{
-				Bucket:       aws.String(key.TargetLogBucketName(customObject)),
-				GrantReadACP: aws.String(key.LogDeliveryURI),
-				GrantWrite:   aws.String(key.LogDeliveryURI),
+			g.Go(func() error {
+				i := &s3.PutBucketAclInput{
+					Bucket:       aws.String(key.TargetLogBucketName(customObject)),
+					GrantReadACP: aws.String(key.LogDeliveryURI),
+					GrantWrite:   aws.String(key.LogDeliveryURI),
+				}
+
+				_, err = sc.AWSClient.S3.PutBucketAcl(i)
+				if err != nil {
+					return microerror.Mask(err)
+				}
+
+				return nil
 			})
-			if err != nil {
-				return microerror.Mask(err)
-			}
+		}
 
-			_, err = sc.AWSClient.S3.PutBucketLifecycleConfiguration(&s3.PutBucketLifecycleConfigurationInput{
-				Bucket: aws.String(key.TargetLogBucketName(customObject)),
-				LifecycleConfiguration: &s3.BucketLifecycleConfiguration{
-					Rules: []*s3.LifecycleRule{
-						{
-							Expiration: &s3.LifecycleExpiration{
-								Days: aws.Int64(int64(r.accessLogsExpiration)),
+		if bucketInput.IsLoggingBucket {
+			g.Go(func() error {
+				i := &s3.PutBucketLifecycleConfigurationInput{
+					Bucket: aws.String(key.TargetLogBucketName(customObject)),
+					LifecycleConfiguration: &s3.BucketLifecycleConfiguration{
+						Rules: []*s3.LifecycleRule{
+							{
+								Expiration: &s3.LifecycleExpiration{
+									Days: aws.Int64(int64(r.accessLogsExpiration)),
+								},
+								Filter: &s3.LifecycleRuleFilter{},
+								ID:     aws.String(LifecycleLoggingBucketID),
+								Status: aws.String("Enabled"),
 							},
-							Filter: &s3.LifecycleRuleFilter{},
-							ID:     aws.String(LifecycleLoggingBucketID),
-							Status: aws.String("Enabled"),
 						},
 					},
-				},
+				}
+
+				_, err = sc.AWSClient.S3.PutBucketLifecycleConfiguration(i)
+				if err != nil {
+					return microerror.Mask(err)
+				}
+
+				return nil
 			})
-			if err != nil {
-				return microerror.Mask(err)
-			}
 		}
 
 		if bucketInput.IsLoggingEnabled {
-			_, err = sc.AWSClient.S3.PutBucketLogging(&s3.PutBucketLoggingInput{
-				Bucket: aws.String(bucketInput.Name),
-				BucketLoggingStatus: &s3.BucketLoggingStatus{
-					LoggingEnabled: &s3.LoggingEnabled{
-						TargetBucket: aws.String(key.TargetLogBucketName(customObject)),
-						TargetPrefix: aws.String(bucketInput.Name + "/"),
+			g.Go(func() error {
+				i := &s3.PutBucketLoggingInput{
+					Bucket: aws.String(bucketInput.Name),
+					BucketLoggingStatus: &s3.BucketLoggingStatus{
+						LoggingEnabled: &s3.LoggingEnabled{
+							TargetBucket: aws.String(key.TargetLogBucketName(customObject)),
+							TargetPrefix: aws.String(bucketInput.Name + "/"),
+						},
 					},
-				},
+				}
+
+				_, err = sc.AWSClient.S3.PutBucketLogging(i)
```

Comment:
The e2e tests are all fine as it seems, but can you confirm the bucket configurations are safe to be applied in parallel @pipo02mix?

Reply:
yeah I think so, they are independent from each other